gisMetadataProcessing.py

```diff
@@ -38,23 +38,36 @@ def splitkeywords(fieldname, newkeyword):
         creatkeyword(value, newkeyword)
 
 
-currdir = os.getcwd()
+def createfile(file, outputfilepath):
+    """Create XML file."""
+    updatedfile = os.path.join(outputfilepath, filename)
+    f = open(updatedfile, 'wb')
+    f.write(etree.tostring(file, pretty_print=True))
+
 
 vocabdict = {'iso': 'ISO 19115 Topic Category', 'fst': 'searchFAST'}
 
 csvfile = 'gis.csv'
+outputfilepath = ''
 
-metadatacsv = csv.DictReader(open(csvfile))
 
+metadatacsv = csv.DictReader(open(csvfile))
 for row in metadatacsv:
     newkeywords = etree.Element('keywords')
     for fieldname, value in row.items():
         if fieldname == 'filename':
-            filename = row[fieldname]
+            filepath = row[fieldname]
+            if '\\' in filepath:
```

Comment:
What are the cases these checks are trying to address?

Reply:
I rethought the workflow based on her feedback. I had initially assumed the script would be in the same directory as the files but there's a complicated directory structure, so she will be putting the full file path into the input CSV rather than just the CSV. I added this to accommodate Windows and non-Windows file paths

Reply:
Ah, ok. In that case, I would suggest using the `pathlib` module. You can replace this entire if/else with something like: 
```python
p = pathlib.Path(filepath)
print(p.name)
```
This should work on both unix and windows.

Reply:
Beautiful! I figured there was a better solution. I'll update this

Reply:
Fixed, thanks!